src/commands/addSshKey.ts

```diff
@@ -34,14 +34,20 @@ export async function addSshKey(context: IActionContext, node?: VirtualMachineTr
         // the VMAccessForLinux extension is necessary to configure more SSH keys
         vmExtension = await computeClient.virtualMachineExtensions.get(node.resourceGroup, node.name, extensionName);
     } catch (e) {
-        vmExtension = { location: vm.location, publisher: 'Microsoft.OSTCExtensions', virtualMachineExtensionType: 'VMAccessForLinux', type: 'Microsoft.Compute/virtualMachines/extensions', typeHandlerVersion: '1.4', autoUpgradeMinorVersion: true };
-        vmExtension.protectedSettings = {
-            ssh_key: (await fse.readFile(sshPublicKey.fsPath)).toString(),
-            // tslint:disable-next-line: strict-boolean-expressions
-            username: vm.osProfile && vm.osProfile.adminUsername || 'azureuser'
-        };
+        const parsedError: IParsedError = parseError(e);
+        if (parsedError.errorType === 'ResourceNotFound') {
+            vmExtension = { location: vm.location, publisher: 'Microsoft.OSTCExtensions', virtualMachineExtensionType: 'VMAccessForLinux', type: 'Microsoft.Compute/virtualMachines/extensions', typeHandlerVersion: '1.4', autoUpgradeMinorVersion: true };
+        }
+
+        throw e;
```

Comment:
`throw e` needs to be in an `else` case otherwise the code in the `if` case won't matter